@stuartwdouglas It failed on Windows with: [ERROR]   ClassLoadingInterruptTestCase.testClassLoaderWhenThreadInterrupted:28 expected: not equal but was: <class io.quarkus.bootstrap.classloader.InterruptClass>

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -101,7 +101,7 @@
         <wildfly-elytron.version>1.11.3.Final</wildfly-elytron.version>
         <jboss-modules.version>1.8.7.Final</jboss-modules.version>
         <jboss-threads.version>3.1.0.Final</jboss-threads.version>
-        <vertx.version>3.8.5</vertx.version>
+        <vertx.version>3.9.0</vertx.version>
         <httpclient.version>4.5.12</httpclient.version>
         <httpcore.version>4.4.13</httpcore.version>
         <httpasync.version>4.1.4</httpasync.version>
@@ -122,13 +122,13 @@
         <infinispan.version>10.1.5.Final</infinispan.version>
         <infinispan.protostream.version>4.3.2.Final</infinispan.protostream.version>
         <caffeine.version>2.8.0</caffeine.version>
-        <netty.version>4.1.45.Final</netty.version>
+        <netty.version>4.1.48.Final</netty.version>
         <reactive-streams.version>1.0.3</reactive-streams.version>
         <test-containers.version>1.12.4</test-containers.version>
         <jboss-logging.version>3.3.2.Final</jboss-logging.version>
         <mutiny.version>0.4.4</mutiny.version>
-        <axle-client.version>0.0.13</axle-client.version>
-        <mutiny-client.version>0.0.13</mutiny-client.version>
+        <axle-client.version>0.0.15</axle-client.version>
+        <mutiny-client.version>0.0.15</mutiny-client.version>
         <kafka2.version>2.5.0</kafka2.version>
         <debezium.version>1.1.1.Final</debezium.version>
         <zookeeper.version>3.5.7</zookeeper.version>

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -34,9 +34,9 @@
            what we work with by self downloading it: -->
         <graal-sdk.version-for-documentation>19.3.1</graal-sdk.version-for-documentation>
         <rest-assured.version>4.1.1</rest-assured.version>
-        <axle-client.version>0.0.12</axle-client.version>
-        <mutiny-client.version>0.0.13</mutiny-client.version>
-        <vertx.version>3.8.5</vertx.version>
+        <axle-client.version>0.0.15</axle-client.version>
+        <mutiny-client.version>0.0.15</mutiny-client.version>
+        <vertx.version>3.9.0</vertx.version>
 
         <!-- Dev tools -->
         <freemarker.version>2.3.30</freemarker.version>

--- a/docs/src/main/asciidoc/getting-started-reactive.adoc
+++ b/docs/src/main/asciidoc/getting-started-reactive.adoc
@@ -454,7 +454,7 @@ public class Fruit {
     }
 
     public static Multi<Fruit> findAll(PgPool client) {
-        return client.query("SELECT id, name FROM fruits ORDER BY name ASC")
+        return client.query("SELECT id, name FROM fruits ORDER BY name ASC").execute()
                 // Create a Multi from the set of rows:
                 .onItem().produceMulti(set -> Multi.createFrom().items(() -> StreamSupport.stream(set.spliterator(), false)))
                 // For each row create a fruit instance
@@ -462,23 +462,23 @@ public class Fruit {
     }
 
     public static Uni<Fruit> findById(PgPool client, Long id) {
-        return client.preparedQuery("SELECT id, name FROM fruits WHERE id = $1", Tuple.of(id))
+        return client.preparedQuery("SELECT id, name FROM fruits WHERE id = $1", Tuple.of(id)).execute()
                 .onItem().apply(RowSet::iterator)
                 .onItem().apply(iterator -> iterator.hasNext() ? from(iterator.next()) : null);
     }
 
     public Uni<Long> save(PgPool client) {
-        return client.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING (id)", Tuple.of(name))
+        return client.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING (id)", Tuple.of(name)).execute()
                 .onItem().apply(pgRowSet -> pgRowSet.iterator().next().getLong("id"));
     }
 
     public Uni<Boolean> update(PgPool client) {
-        return client.preparedQuery("UPDATE fruits SET name = $1 WHERE id = $2", Tuple.of(name, id))
+        return client.preparedQuery("UPDATE fruits SET name = $1 WHERE id = $2", Tuple.of(name, id)).execute()
                 .onItem().apply(pgRowSet -> pgRowSet.rowCount() == 1);
     }
 
     public static Uni<Boolean> delete(PgPool client, Long id) {
-        return client.preparedQuery("DELETE FROM fruits WHERE id = $1", Tuple.of(id))
+        return client.preparedQuery("DELETE FROM fruits WHERE id = $1", Tuple.of(id)).execute()
                 .onItem().apply(pgRowSet -> pgRowSet.rowCount() == 1);
     }
 
@@ -541,12 +541,12 @@ public class FruitResource {
     }
 
     private void initdb() {
-        client.query("DROP TABLE IF EXISTS fruits")
-                .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)"))
-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Kiwi')"))
-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Durian')"))
-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pomelo')"))
-                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Lychee')"))
+        client.query("DROP TABLE IF EXISTS fruits").execute()
+                .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)").execute())
+                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Kiwi')").execute())
+                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Durian')").execute())
+                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pomelo')").execute())
+                .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Lychee')").execute())
                 .await().indefinitely();
     }
 

--- a/docs/src/main/asciidoc/reactive-sql-clients.adoc
+++ b/docs/src/main/asciidoc/reactive-sql-clients.adoc
@@ -184,13 +184,20 @@ It returns a `Uni` and thus can be composed to execute queries sequentially:
 
 [source,java]
 ----
- client.query("DROP TABLE IF EXISTS fruits")
-    .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)"))
-    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')"))
-    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')"))
-    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')"))
+ client.query("DROP TABLE IF EXISTS fruits").execute()
+    .flatMap(r -> client.query("CREATE TABLE fruits (id SERIAL PRIMARY KEY, name TEXT NOT NULL)").execute())
+    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')").execute())
+    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')").execute())
+    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')").execute())
     .await().indefinitely();
 ----
+
+[IMPORTANT]
+.Breaking Change in Quarkus 1.5
+====
+Vert.x 3.9, integrated in Quarkus 1.5, introduces a breaking change regarding the `query` methods.
+To retrieve the result you need to call `.execute()`.
+====
 
 NOTE: Wondering why we need block until the latest query is completed?
 This code is part of a `@PostConstruct` method and Quarkus invokes it synchronously.
@@ -209,7 +216,7 @@ To retrieve all the data, we will use the `query` method again:
 
 [source,java]
 ----
-Uni<RowSet> rowSet = client.query("SELECT id, name FROM fruits ORDER BY name ASC");
+Uni<RowSet> rowSet = client.query("SELECT id, name FROM fruits ORDER BY name ASC").execute();
 ----
 
 When the operation completes, we will get a `RowSet` that has all the rows buffered in memory.
@@ -244,7 +251,7 @@ Putting it all together, the `Fruit.findAll` method looks like:
 .src/main/java/org/acme/vertx/Fruit.java
 ----
     public static Uni<List<Fruit>> findAll(PgPool client) {
-        return client.query("SELECT id, name FROM fruits ORDER BY name ASC")
+        return client.query("SELECT id, name FROM fruits ORDER BY name ASC").execute()
             .map(pgRowSet -> {
                 List<Fruit> list = new ArrayList<>(pgRowSet.size());
                 for (Row row : pgRowSet) {
@@ -288,7 +295,7 @@ The Reactive PostgreSQL Client can also prepare queries and take parameters that
 
 [source,java]
 ----
-client.preparedQuery("SELECT name FROM fruits WHERE id = $1", Tuple.of(id))
+client.preparedQuery("SELECT name FROM fruits WHERE id = $1", Tuple.of(id)).execute()
 ----
 
 TIP: The SQL string can refer to parameters by position, using $1, $2, ...etc.
@@ -300,7 +307,7 @@ Equipped with this tooling, we are able to safely use an `id` provided by the us
 .src/main/java/org/acme/vertx/Fruit.java
 ----
 public static Uni<Fruit> findById(PgPool client, Long id) {
-    return client.preparedQuery("SELECT id, name FROM fruits WHERE id = $1", Tuple.of(id)) // <1>
+    return client.preparedQuery("SELECT id, name FROM fruits WHERE id = $1", Tuple.of(id)).execute() // <1>
             .map(RowSet::iterator) // <2>
             .map(iterator -> iterator.hasNext() ? from(iterator.next()) : null); // <3>
 }
@@ -331,7 +338,7 @@ The same logic applies when saving a `Fruit`:
 .src/main/java/org/acme/vertx/Fruit.java
 ----
 public Uni<Long> save(PgPool client) {
-    return client.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING (id)", Tuple.of(name))
+    return client.preparedQuery("INSERT INTO fruits (name) VALUES ($1) RETURNING (id)", Tuple.of(name)).execute()
             .map(pgRowSet -> pgRowSet.iterator().next().getLong("id"));
 }
 ----
@@ -362,7 +369,7 @@ Let's use this to support removal of fruits in the database:
 .src/main/java/org/acme/vertx/Fruit.java
 ----
 public static Uni<Boolean> delete(PgPool client, Long id) {
-    return client.preparedQuery("DELETE FROM fruits WHERE id = $1", Tuple.of(id))
+    return client.preparedQuery("DELETE FROM fruits WHERE id = $1", Tuple.of(id)).execute()
             .map(pgRowSet -> pgRowSet.rowCount() == 1); // <1>
 }
 ----

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
@@ -62,6 +62,7 @@ import io.vertx.core.Handler;
 import io.vertx.core.Verticle;
 import io.vertx.core.Vertx;
 import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
 import io.vertx.core.http.HttpServer;
 import io.vertx.core.http.HttpServerOptions;
 import io.vertx.core.http.HttpServerRequest;
@@ -944,7 +945,11 @@ public class VertxHttpRecorder {
                                 //this can happen if blocking authentication is involved for get requests
                                 if (!event.request().isEnded()) {
                                     event.request().resume();
-                                    bodyHandler.handle(event);
+                                    if (CAN_HAVE_BODY.contains(event.request().method())) {
+                                        bodyHandler.handle(event);
+                                    } else {
+                                        event.next();
+                                    }
                                 } else {
                                     event.next();
                                 }
@@ -955,9 +960,16 @@ public class VertxHttpRecorder {
                     });
                 } else {
                     event.request().resume();
-                    bodyHandler.handle(event);
+                    if (CAN_HAVE_BODY.contains(event.request().method())) {
+                        bodyHandler.handle(event);
+                    } else {
+                        event.next();
+                    }
                 }
             }
         };
     }
+
+    private static final List<HttpMethod> CAN_HAVE_BODY = Arrays.asList(HttpMethod.POST, HttpMethod.PUT, HttpMethod.PATCH,
+            HttpMethod.DELETE);
 }

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/QuarkusRequestWrapper.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/filters/QuarkusRequestWrapper.java
@@ -9,6 +9,7 @@ import org.jboss.logging.Logger;
 import io.quarkus.vertx.http.runtime.AbstractRequestWrapper;
 import io.vertx.core.Handler;
 import io.vertx.core.http.Cookie;
+import io.vertx.core.http.CookieSameSite;
 import io.vertx.core.http.HttpServerRequest;
 import io.vertx.core.http.HttpServerResponse;
 
@@ -197,6 +198,11 @@ public class QuarkusRequestWrapper extends AbstractRequestWrapper {
 
         @Override
         public Cookie setHttpOnly(boolean httpOnly) {
+            return null;
+        }
+
+        @Override
+        public Cookie setSameSite(CookieSameSite policy) {
             return null;
         }
 

